types/defines/vectorize.d.ts

```diff
@@ -59,13 +59,11 @@ type VectorizeDistanceMetric = "euclidean" | "cosine" | "dot-product";
  */
 type VectorizeMetadataRetrievalLevel = "all" | "indexed" | "none";
 
-interface VectorizeQueryOptions<
-  MetadataReturn extends boolean | VectorizeMetadataRetrievalLevel = boolean,
-> {
```

Comment:
This is a breaking change to the types—is it absolutely needed?

Reply:
Agree with @penalosa this is a breaking change. If the problem is the `returnMetadata` field "continued to expect a boolean field for V2 as well instead of VectorizeMetadataRetrievalLevel" can we help you make that work rather than remove the generic entirely?

Reply:
This change should have no effect on the way users would interact with Vectorize V1 using worker bindings. So I am not sure if I would term this as a breaking change.

That being said, I would be happy to understand if there could be any alternative options that could be suitable for our purpose.

Reply:
@RamIdeas @penalosa Do you have any suggestions on what I can try to make generics work, or to have a smaller impact on the type definitions?

Reply:
Garvit and I discussed this on a call and decided this isn't a breaking change to v1 users and the union type (over the generic type) is more ergonomic for both v1 and v2 users (+ incorrect usage is easily caught in js validation at runtime)